### PR TITLE
issue_1613  Test

### PR DESCRIPTION
**Pull request by GonzoSpire**
#1613

### CODE REVIEW

File: Tests/Linq/DataProvider/SqlServerTests.cs [hunk 1]
```diff
@@ -1348,5 +1348,35 @@ public void TestIssue1144([AllSqlServerDataContext] string context)
...
+		{
+			[Column("dt"), Nullable] 
+			public DateTimeOffset? DateTimeOffset { get; set; }
+		}
+		[Test]
+		public void Issue1613Test([AllSqlServerDataContext] string context)
+		{
+			using (var db = GetDataContext(context))
+			{
+				db.CreateTable<Issue1613Table>();
```
Comment (sdanyliv) [inline Tests/Linq/DataProvider/SqlServerTests.cs]: Use CreateLocalTable, which automatically inserts records and drops table.

**Comment by GonzoSpire** [comment]
Add tests that fail with:
 Message: LinqToDB.Linq.LinqException : 'ToNotNull(selectParam.DateTimeOffset).Date' cannot be converted to SQL.
Message: LinqToDB.Linq.LinqException : 'ToNotNull(selectParam.DateTimeOffset).TimeOfDay' cannot be converted to SQL.

File: Tests/Linq/DataProvider/SqlServerTests.cs [hunk 2]
```diff
@@ -1348,5 +1348,69 @@ public void TestIssue1144([AllSqlServerDataContext] string context)
...
+		[Test]
+		public void Issue1613Test1([AllSqlServerDataContext] string context)
+		{
+			using (var db = GetDataContext(context))
+			using (var table = db.CreateLocalTable(GenerateData()))
+			{ 
+
+				var query1 = table.GroupBy(x => x.DateTimeOffset).Select(g => g.Key).ToList();
+				var query2 = table.Select(r => r.DateTimeOffset).ToList();
+
```
Comment (MaceWindu) [inline Tests/Linq/DataProvider/SqlServerTests.cs]: use list comparison assert, checking count is not enough

**Comment by MaceWindu** [comment]
Failing tests probably will be fixed by #780

**Comment by MaceWindu** [comment]
Tests actually fail due to #1666

**Pull request merged**